src/Pos/Lrc/Worker.hs

```diff
@@ -142,23 +145,23 @@ lrcDo getPeers epoch consumers tip = tip <$ do
         Nothing -> throwM UnknownBlocksForLrc
         Just (NewestFirst -> blunds) ->
             withBlocksRolledBack blunds $ do
-                -- We have to compute richmen before calculating the seed because
-                -- seed calculation relies on knowing richmen
                 issuersComputationDo epoch
-                richmenComputationDo epoch consumers
+                -- We have to compute richmen before calculating the seed
+                -- because seed calculation relies on knowing richmen. We
+                -- give 'calcAndStoreSeed' as a callback to
+                -- 'richmenComputationDo' because it's easier than reading
+                -- richmen from the database (I couldn't find a function that
+                -- does it on a locked LRC database)
+                let simpleConsumer = lrcConsumerFromComponentSimple @RCSsc bvdMpcThd
+                let calcSeedConsumer = simpleConsumer {
+                        lcComputedCallback = calcAndStoreSeed }
+                richmenComputationDo epoch (calcSeedConsumer:consumers)
```

Comment:
Does “richmen from 9th epoch” mean “nodes who were rich at the beginning of 9th epoch” or “nodes who were rich at the end (after the last slot) of the 9th epoch”?